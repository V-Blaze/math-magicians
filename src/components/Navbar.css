.navbar {
  display: flex;
  padding: 0.5rem 2rem;
  justify-content: space-between;
  align-items: center;
  background-color: gray;
}

.title {
  font-weight: bold;
  font-size: 3rem;
}

.menu-list {
 display: flex;
 justify-content: space-between;
 list-style: none;
}

.list-item {
 margin-right: 0.5rem;
 font-size: 1.5rem;
 cursor: pointer;
}