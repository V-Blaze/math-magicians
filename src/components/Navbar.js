import React from 'react';

// stylesheet
import './Navbar.css';

const Navbar = () => (
  <div className="navbar">
    <div className="title-div">
      <h2 className="title">Math Magicians</h2>
    </div>
    <div className="menu">
      <ul className="menu-list">
        <li className="list-item">Home</li>
        <li className="list-item">Calculator</li>
        <li className="list-item">Quote</li>
      </ul>
    </div>
  </div>
);

export default Navbar;
