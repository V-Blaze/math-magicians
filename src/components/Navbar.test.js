import { render, screen } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import Navbar from './Navbar';

describe('testing the Navbar', () => {
  it('testing the title inside the Navbar ', () => {
    render(
      <Router>
        <Navbar />
      </Router>,
    );
    const title = screen.getByText(/Math Magicians/i);
    expect(title).toBeInTheDocument();
  });
  it('testing the Link Home inside the Navbar ', () => {
    render(
      <Router>
        <Navbar />
      </Router>,
    );
    const linkElement = screen.getByText(/Home/i);
    expect(linkElement).toBeInTheDocument();
  });
});
