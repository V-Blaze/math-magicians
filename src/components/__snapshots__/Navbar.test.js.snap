// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the Navbar snapshots 1`] = `
<div
  className="navbar"
>
  <div
    className="title-div"
  >
    <h2
      className="title"
    >
      Math Magicians
    </h2>
  </div>
  <div
    className="menu"
  >
    <nav>
      <ul
        className="menu-list"
      >
        <li
          className="list-item"
        >
          <a
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li
          className="list-item"
        >
          <a
            href="/calculator"
            onClick={[Function]}
          >
            Calculator
          </a>
        </li>
        <li
          className="list-item"
        >
          <a
            href="/quote"
            onClick={[Function]}
          >
            Quote
          </a>
        </li>
      </ul>
    </nav>
  </div>
</div>
`;
